services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/AutoRerouteV2Spec.groovy

```diff
@@ -444,6 +407,90 @@ class AutoRerouteV2Spec extends HealthCheckSpecification {
+        def switchPair = topologyHelper.getAllNeighboringSwitchPairs().find {
+            it.paths.findAll { it.size() == 2 }.size() > 1
+        } ?: assumeTrue("No suiting switches found", false)
+
+        //Main and backup paths for further manipulation with them
+        def mainPath = switchPair.paths.min { it.size() }
+        def backupPath = switchPair.paths.findAll { it != mainPath }.min { it.size() }
+        def altPaths = switchPair.paths.findAll { it != mainPath && it != backupPath }
+
+        //All alternative paths are unavailable (bring ports down on the srcSwitch)
+        List<PathNode> broughtDownPorts = []
+        altPaths.unique { it.first() }.each { path ->
+            def src = path.first()
+            broughtDownPorts.add(src)
+            antiflap.portDown(src.switchId, src.portNo)
+        }
+        Wrappers.wait(antiflapMin + WAIT_OFFSET) {
+            assert northbound.getAllLinks().findAll {
+                it.state == IslChangeType.FAILED
+            }.size() == broughtDownPorts.size() * 2
+        }
+
+        //Main path more preferable than the backup
+        pathHelper.makePathMorePreferable(mainPath, backupPath)
+
+        and: "A flow"
+        def flow = flowHelperV2.randomFlow(switchPair)
+        flowHelperV2.addFlow(flow)
+        assert PathHelper.convert(northbound.getFlowPath(flow.flowId)) == mainPath
+
+        when: "Disconnect the src switch from the controller"
+        def islToBreak = pathHelper.getInvolvedIsls(mainPath).first()
+        def islToReroute = pathHelper.getInvolvedIsls(backupPath).first()
+        lockKeeper.knockoutSwitch(switchPair.src)
+        Wrappers.wait(discoveryTimeout + WAIT_OFFSET) {
+            assert northbound.getSwitch(switchPair.src.dpId).state == SwitchChangeType.DEACTIVATED
+        }
+
+        and: "Mark the switch as ACTIVE in db" // just to reproduce #3131
+        database.setSwitchStatus(switchPair.src.dpId, SwitchStatus.ACTIVE)
+
+        and: "Init auto reroute (bring ports down on the dstSwitch)"
+        antiflap.portDown(islToBreak.dstSwitch.dpId, islToBreak.dstPort)
+
+        then: "Flow is not rerouted and flow status is 'Down'"
+        TimeUnit.SECONDS.sleep(rerouteDelay * 2) // it helps to be sure that the auto-reroute operation is completed
+        Wrappers.wait(WAIT_OFFSET / 2) {
+            assert northbound.getLink(islToBreak).state == FAILED
+            // just to be sure that backup ISL is not failed
+            assert northbound.getLink(islToReroute).state == DISCOVERED
+            assert northbound.getFlowStatus(flow.flowId).status == FlowState.DOWN
+            assert PathHelper.convert(northbound.getFlowPath(flow.flowId)) == mainPath
+        }
+
+        when: "Connect the switch back to the controller"
+        database.setSwitchStatus(switchPair.src.dpId, SwitchStatus.INACTIVE) // set real status
+        lockKeeper.reviveSwitch(switchPair.src)
```

Comment:
While we don't have a history verification that the reroute reason was 'switch up event', I'd like to see an explicit assert that all the switch-related isls are up at this point (or at least one?). Otherwise - I'm not sure what was the reason of reroute and it is very easy to receive a false-successful run.

Reply:
I thought `L464` should be enough.

I can add the same verification before activating the switch, but I think there is no sense
 it takes system less than second for getting from `L464` to `L471`

Reply:
OK, didn't realize that `islToReroute` belongs to switch to break. It's fine then